### PR TITLE
[Feature] MS/MS precursor filtering by mass list

### PR DESCRIPTION
**Pull request by cbielow**
additional option '-peak_options:pc_mz_list' for TOPP:FileFilter allows to provide a mass list.
All MS2 spectra whose precursor covers any of the given values is retained.
Other MS2 spectra are discarded.

Also:
 - renamed precursor mass range filter from '-pc_mz' to '-peak_options:pc_mz_range'

### CODE REVIEW

File: src/openms/include/OpenMS/KERNEL/RangeUtils.h [hunk 1]
```diff
@@ -652,6 +653,61 @@ namespace OpenMS
...
+  {
+
+public:
+    /**
+      @brief Constructor
+
+      @param vec_mz Vector of m/z values, of which at least one needs to be covered
+      @param reverse if @p reverse is true, operator() returns true if the isolation window is outside for ALL m/z values.
+    */
+    IsInIsolationWindow(std::vector<double> vec_mz, bool reverse = false) :
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/KERNEL/RangeUtils.h]: instead of reverse maybe one could use: std::not1 from <functional> ?

**Comment by timosachsenberg** [comment]
changing the parameter name is a "breaking change" for most workflows so we should document this in the CHANGELOG

**Comment by timosachsenberg** [comment]
I like it. Maybe in the future it would be also nice to have the same function for the uncharged mass...

File: src/openms/include/OpenMS/KERNEL/RangeUtils.h (hunk 1 above)
Comment (hroest) [inline src/openms/include/OpenMS/KERNEL/RangeUtils.h]: it seems less confusing right now, I like "reverse"

Comment (timosachsenberg) [inline src/openms/include/OpenMS/KERNEL/RangeUtils.h]: Fine with me

**Comment by cbielow** [comment]
right.
But we should compare INI files prior to release for ALL TOPP tools and 
make a list. I think Hannes has done that before.

Am 2/23/2016 um 6:24 PM schrieb Timo Sachsenberg:

**Comment by timosachsenberg** [comment]
Renames must be manually annotated (from add and remove entries) so a note might save some time verifying this. But I guess we can go without it. 
----- Original Message -----

**Pull request merged**